pandas/core/dtypes/cast.py

```diff
@@ -723,8 +725,19 @@ def astype_nansafe(arr, dtype, copy=True, skipna=False):
 
     elif is_object_dtype(arr):
 
-        # work around NumPy brokenness, #1987
         if np.issubdtype(dtype.type, np.integer):
+            # TODO: this is an old numpy compat branch that is not necessary
```

Comment:
Yikes.  Not sure off the top whats going on here.

Why the comment saying it is no longer necessary?  np.timedelta64 is still a subdtype of np.integer

To the extent that this branch is just for timedelta64 dtypes, I would make that explicit instead of checking `if np.issubdtype(dtype.type, np.integer)`